target/linux/ramips/dts/mt7628an_tplink_8m.dtsi

```diff
@@ -66,5 +66,8 @@
 
 &ethernet {
 	mtd-mac-address = <&factory 0xf100>;
-	mediatek,portmap = "llllw";
+};
+
+&esw {
+	mediatek,portmap = <0x2f>;
```

Comment:
Several devices like c20-v4 and c50-v3 using this DTSI are configured with 0:wan and 4:lan here. Others are configured as one-port.

To me it looks like this worked mostly because the "llllw" was not applied before. We thus might break them by putting the correct mediatek,portmap and I would vote for removing it from the DTSI and setting it on a per-device base (in DTS) here.

I'd personally set the following based on 02_network:
	tplink,archer-c20-v4|\
	tplink,archer-c50-v3|\
	tplink,tl-mr3420-v5|\
	tplink,tl-wr840n-v4|\
	tplink,tl-wr841n-v13|\
	tplink,tl-wr842n-v5|\
Those have 0:wan and 4:lan and thus should be set to 0x3e.
```
	tplink,tl-wr902ac-v3)
		ucidef_add_switch "switch0" \
			"4:lan" "6@eth0"
		;;
```
This one we might set to 0x3e as well or just not set it at all.

The remaining tplink_8m are one-ports where I would not put a setting, as I cannot not see which would be correct.

8m-split-uboot seems to be correct so far.

Reply:
Thanks for checking it out. I should have done it earlier...
I've checked the all devices in the commit and I found your description is correct.
I updated the patch and it should have resolved the all stuff you mentioned.